beacon-chain/rpc/beaconv1/validator_test.go

```diff
@@ -0,0 +1,316 @@
+		ids := [][]byte{pubkey1[:], []byte("90"), pubkey3[:], []byte("129")}
+		resp, err := s.ListValidators(ctx, &ethpb.StateValidatorsRequest{
+			StateId: []byte("head"),
+			Id:      ids,
+		})
+		require.NoError(t, err)
+		for i, val := range resp.Data {
+			assert.Equal(t, idNums[i], val.Index)
+			assert.Equal(t, true, bytes.Equal(pubkeys[i], val.Validator.PublicKey))
+		}
+	})
+}
+
+func TestListValidatorBalances(t *testing.T) {
+	ctx := context.Background()
+
+	var state iface.BeaconState
+	state, _ = testutil.DeterministicGenesisState(t, 8192)
+
+	t.Run("Head List Validators Balance by index", func(t *testing.T) {
+		s := Server{
+			StateFetcher: statefetcher.StateProvider{
+				ChainInfoFetcher: &chainMock.ChainService{State: state},
+			},
+		}
+
+		ids := [][]byte{[]byte("15"), []byte("26"), []byte("400")}
+		idNums := []types.ValidatorIndex{15, 26, 400}
+		resp, err := s.ListValidatorBalances(ctx, &ethpb.ValidatorBalancesRequest{
+			StateId: []byte("head"),
+			Id:      ids,
+		})
+		require.NoError(t, err)
+		for i, val := range resp.Data {
+			assert.Equal(t, idNums[i], val.Index)
+			assert.Equal(t, params.BeaconConfig().MaxEffectiveBalance, val.Balance)
+		}
+	})
+
+	t.Run("Head List Validators Balance by pubkey", func(t *testing.T) {
+		s := Server{
+			StateFetcher: statefetcher.StateProvider{
+				ChainInfoFetcher: &chainMock.ChainService{State: state},
+			},
+		}
+		idNums := []types.ValidatorIndex{20, 66, 90, 100}
+		pubkey1 := state.PubkeyAtIndex(types.ValidatorIndex(20))
+		pubkey2 := state.PubkeyAtIndex(types.ValidatorIndex(66))
+		pubkey3 := state.PubkeyAtIndex(types.ValidatorIndex(90))
+		pubkey4 := state.PubkeyAtIndex(types.ValidatorIndex(100))
+		pubKeys := [][]byte{pubkey1[:], pubkey2[:], pubkey3[:], pubkey4[:]}
+		resp, err := s.ListValidatorBalances(ctx, &ethpb.ValidatorBalancesRequest{
+			StateId: []byte("head"),
+			Id:      pubKeys,
+		})
+		require.NoError(t, err)
+		for i, val := range resp.Data {
+			assert.Equal(t, idNums[i], val.Index)
+			assert.Equal(t, params.BeaconConfig().MaxEffectiveBalance, val.Balance)
+		}
+	})
+
+	t.Run("Head List Validators Balance by both index and pubkey", func(t *testing.T) {
+		s := Server{
+			StateFetcher: statefetcher.StateProvider{
+				ChainInfoFetcher: &chainMock.ChainService{State: state},
+			},
+		}
+
+		idNums := []types.ValidatorIndex{20, 90, 170, 129}
+		pubkey1 := state.PubkeyAtIndex(types.ValidatorIndex(20))
+		pubkey3 := state.PubkeyAtIndex(types.ValidatorIndex(170))
+		ids := [][]byte{pubkey1[:], []byte("90"), pubkey3[:], []byte("129")}
+		resp, err := s.ListValidatorBalances(ctx, &ethpb.ValidatorBalancesRequest{
+			StateId: []byte("head"),
+			Id:      ids,
+		})
+		require.NoError(t, err)
+		for i, val := range resp.Data {
+			assert.Equal(t, idNums[i], val.Index)
+			assert.Equal(t, params.BeaconConfig().MaxEffectiveBalance, val.Balance)
+		}
+	})
+}
+
+func TestListCommittees(t *testing.T) {
```

Comment:
Please also write a test for fetching committees for a given epoch.

Reply:
Tried to add... was very difficult. Need some help since it touches so many parts of state.

Reply:
Thanks for adding it! Couldn't get it to work for some reason